test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,191 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+				return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName[0], err)
+			}
+
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIP.String() == ipAddress.IPAddress {
+					return false, nil
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func checkOVSPorts(before, after []string) bool {
+	if len(before) != len(after) {
+		return false
+	}
+	afterSet := make(map[string]struct{})
+	for _, port := range after {
+		afterSet[port] = struct{}{}
+	}
+
+	for _, port := range before {
+		if _, exists := afterSet[port]; !exists {
+			return false
+		}
+	}
+
+	return true
+}
+
+func checkIPsAndIfaces(before, after map[string]net.IP) bool {
+	if len(before) != len(after) {
+		return false
+	}
+	for iface, beforeIP := range before {
+		afterIP, exists := after[iface]
+		if !exists || !beforeIP.Equal(afterIP) {
+			return false
+		}
+	}
+	return true
+}
+
+func (data *testData) getOVSPortsOnSecondaryBridge(t *testing.T, nodeName string) ([]string, error) {
+	cmd := []string{"ovs-vsctl", "list-ports", "br-secondary"}
+
+	stdout, stderr, err := data.e2eTestData.RunCommandFromAntreaPodOnNode(nodeName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ovs-vsctl on node %s: %v\nstderr: %s", nodeName, err, stderr)
+	}
+
+	ovsPorts := strings.Fields(stdout)
+	return ovsPorts, nil
+}
+
+// reconcilationAfterAgentRestart verifies OVS Ports cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	for _, pod := range data.pods {
+		ovsPortsBeforeAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, pod.nodeName)
+		if err != nil {
+			return fmt.Errorf("failed to get Secondary bridge OVS Ports before agent restart: %w", err)
+		}
+
+		beforeAgentRestartIPsAndIfaces, err := data.listPodIPs(pod)
+		if err != nil {
+			return fmt.Errorf("failed to get Pod IP before agent restart: %w", err)
+		}
+
+		// Restarting the Antrea agent
+		if err := data.e2eTestData.RestartAntreaAgentPods(30 * time.Second); err != nil {
+			t.Fatalf("Failed to restart Antrea agent Pods: %v", err)
+		}
+
+		ovsPortsAfterAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, pod.nodeName)
+		if err != nil {
+			return fmt.Errorf("failed to get Secondary bridge OVS Ports after agent restart: %w", err)
+		}
+
+		afterAgentRestartIPsAndIfaces, err := data.listPodIPs(pod)
+		if err != nil {
+			return fmt.Errorf("failed to get Pod IP after agent restart: %w", err)
+		}
+
+		if !checkOVSPorts(ovsPortsBeforeAgentRestart, ovsPortsAfterAgentRestart) {
+			return fmt.Errorf("Secondary bridge OVS Ports mismatch after agent restart: %w", err)
+		}
+
+		if !checkIPsAndIfaces(beforeAgentRestartIPsAndIfaces, afterAgentRestartIPsAndIfaces) {
+			return fmt.Errorf("Pod: %s, Interfaces and IPs mismatch after agent restart", pod.podName)
+		}
+	}
+
+	return data.deleteVLANPod(t)
+}
+
+func (data *testData) deleteVLANPod(t *testing.T) error {
```

Comment:
IMO you should merge this function into `reconcilationAfterAgentRestart`. It doesn't really make sense as a standalone function with such a generic name. If you switch to assertions as I suggested above, the functions will be shorter anyway.

Reply:
updated